### DISCUSSION TITLE
Quicker feedback loop with encrypted credentials

### DISCUSSION THREAD

**Discussion by jmctune** [renovatebot/renovate#20738] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

My `hostRules` have encrypted credentials to access a private docker repository in ECR. To enable a quicker feedback loop when testing changes to my `renovate.json` file, I'm attempting to run renovate locally via the provided Docker image.

I'm curious if there's a way to override what's in the remote with locally provided values to the docker container at runtime as my local environment cannot decrypt the encrypted credentials that were encrypted with this tool: https://app.renovatebot.com/encrypt.

Example of my hostRules:

```json
{
  "hostRules": [
    {
      "hostType": "docker",
      "matchHost": "999999999999.dkr.ecr.us-east-1.amazonaws.com",
      "encrypted": {
        "username": "encrypted_string",
        "password": "encrypted_string"
      }
    }
  ]
}
```

### What you tried so far.

N/A. I've tried to find this in the documentation, but nothing has stood out to me.

### Relevant debug logs

<details><summary>Logs</summary>

```
ERROR: Found encrypted data but no privateKey
```

</details>

**Comment by rarkins** [comment, marked as answer]
Forking or importing into a separate repo to test with is usually easiest

**Reply by jmctune** [reply]
This would imply the need to commit the **raw** credentials to Github though, correct?

**Reply by rarkins** [reply]
No, you can do this:
- Fork or import the repo into a new repo
- Delete the hostRules with credentials from the new repo
- Define the same hostRules - but without encrypting - locally in a `config.js` file
- Run Renovate locally. It will combine your local rules with the config it finds in the repo
